include/libcyphal/transport/dynamic_buffer.hpp

```diff
@@ -0,0 +1,120 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_DYNAMIC_BUFFER_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_DYNAMIC_BUFFER_HPP_INCLUDED
+
+#include <cetl/any.hpp>
+
+#include <cstdint>
+
+namespace libcyphal
+{
+namespace transport
+{
+
+/// The buffer is movable but not copyable because copying the contents of a buffer is considered wasteful.
+/// The buffer behaves as if it's empty if the underlying implementation is moved away.
+///
+class DynamicBuffer final
```

Comment:
We're circling back to the same conversation again: we do in fact need this DynamicBuffer because it is designed to provide a contiguous view of fragmented lizard-specific buffers. It is not flat. It erases the type information of the underlying buffer and makes it appear contiguous. Please refer here for details: https://github.com/OpenCyphal-Garage/libcyphal/blob/main/docs/design/readme.md#dynamicbuffer

Reply:
How about `FragmentedBuffer` or `ScatteredBuffer`?

Reply:
How about `FragmentedBuffer` or `ScatteredBuffer`?

Reply:
Fragmented is a dirty word so let's go with ScatteredBrain, er, ScatteredBuffer

Reply:
I'll change it to →  `ScatteredBuffer`

Reply:
done